src/dscanner/utils.d

```diff
@@ -105,7 +106,7 @@ string[] expandArgs(string[] args)
 		else
 			foreach (item; dirEntries(arg, SpanMode.breadth).map!(a => a.name))
 			{
-				if (isFileSafe(item) && (item.endsWith(`.d`) || item.endsWith(`.di`)))
+				if (isFileSafe(item) && (item.endsWith(`.d`) || item.endsWith(`.di`)) && !item.canFind(dirSeparator ~ '.'))
```

Comment:
wouldn't the `dirSeparator ~ '.'` allocate a lot of garbage? I think it should be put in a constant outside the loops.

I can't test if it does though because run.dlang.io doesn't support emitting the profile=gc log and ldc doesn't support it either and dmd isn't downloadable here.

Reply:
ok nvm could test now and turns out it doesn't even allocate with dirSeparator

Reply:
I would be very surprised if the compiler couldn't constant fold that.

Reply:
Aye. `dirSeperator` is an enum constant. `"."` is a compile time constant. I've never read the compiler source, but I'd assume it's a compile time constant and treated as such.